### PR TITLE
feat(hydroflow_plus): switch launch to be a macro to enable borrowing locals

### PR DESCRIPTION
**Pull request by shadaj**
feat(hydroflow_plus): switch launch to be a macro to enable borrowing locals

Summary:

Test Plan:

### CODE REVIEW

**Comment by shadaj** [comment]
Ran into trouble borrowing things outside the `launch` closure while implementing profiling counters with @davidchuyaya. Perhaps there is a fix by putting the right lifetimes in the right places, but I cannot figure them out so this works in the meantime. Maybe @MingweiSamuel knows a proper solution.

**Comment by MingweiSamuel** [comment]
Not really sure exactly where you mean in this PR, for the lifetime stuff

File: hydroflow/src/util/cli.rs
Comment (MingweiSamuel) [inline hydroflow/src/util/cli.rs:1]: ^

I think I might be able to make the `fn` version work once I can reproduce the error

Comment (MingweiSamuel) [reply hydroflow/src/util/cli.rs:1]: Ok, couldn't get this to work, may be possible with some sort of multiple-layered closures with scoped threads but not worth the complexity

Comment (shadaj) [reply hydroflow/src/util/cli.rs:1]: 😢

**Review by MingweiSamuel — APPROVED** [review]

**Pull request merged**